vllm/_core_ext.pyi

```diff
@@ -0,0 +1,132 @@
+from typing import Optional, Union
+
+class ScalarType:
+    """
+    ScalarType can represent a wide range of floating point and integer types,
+    in particular it can be used to represent sub-byte data types (something
+    that torch.dtype currently does not support). It is also cabaable of 
+    representing types with a bias, i.e. the stored_value = value + bias, this
+    is useful for quantized types (e.g. standard GPTQ 4bit uses a bias of 8).
+    """
```

Comment:
Maybe add a reference to the C++ code here (and vice versa), so it is more obvious that both of these files need to be kept in sync?